app/lib/actions/foreman_ansible/play_host_roles.rb

```diff
@@ -0,0 +1,56 @@
+module Actions
+  module ForemanAnsible
+    # Actions that initiaztes the playbook run for roles assigned to
+    # the host. It doest that either locally or via a proxy when available.
+    class PlayHostRoles < Actions::EntryAction
+      include ::Actions::Helpers::WithContinuousOutput
+      include ::Actions::Helpers::WithDelegatedAction
+
+      def plan(host, proxy_selector = ::ForemanAnsible::ProxySelector.new)
+        input[:host] = { :id => host.id, :name => host.fqdn }
+        proxy = proxy_selector.determine_proxy(host)
```

Comment:
The behaviour right now is:

1. if there is a proxy available for the host with ansible feature and it's running, it tries use it
2. if there a proxy available for the host with ansible feature, but it's not running, fail, because the user probably expected to use the proxy
3. if there is no proxy available with ansible feature, use the build in

It's not this way to address the simple-to start case, while respecting requirements on more complex network topologies and production system, where it's more preferred to use the proxy. For example, if I have some remote location with a proxy and hosts are not reachable from the main server, I would not expect the main server to try running directly, even if the proxy is down at the moment.